### DISCUSSION TITLE
Request for Parallel Loading of Editors in Form File

### DISCUSSION THREAD

**Discussion by EdwinThomas11** [serenity-is/Serenity#7121] [Help]
Hi Team,

I have a form file with multiple properties, some of which have editors. Currently, all editors, including lookup editors, are loading sequentially. I would like them to load simultaneously instead. Your assistance with this matter would be greatly appreciated @VictorTomaili .

Please refer to the provided code snippet for context:

[MyLookupEditor]
public String LookUpEditor { get; set; }

[MyIdsEditor]
public List<Int64> MyIds { get; set; }

[FirstRowsEditor]
public List<FirstRow> FirstRows { get; set; }

[SecondRowsEditor]
public List<SecondRow> SecondRows { get; set; }

[HtmlNoteContentEditor(Rows = 4), FullWidth]
public String MyHtmlContentEditor { get; set; }

Best regards,
Edwin Thomas D

**Comment by VictorTomaili** [comment]
use async option or servicelookupeditors

**Reply by EdwinThomas11** [reply]
I'm unable to change it to the ServiceLookupEditors because these fields belong to different rows that are displayed as a grid. I attempted to implement asynchronous loading, but the issue persists as the properties are not loaded in the default.MyForm.ts file, which prevents me from utilising those properties in my dialog files.

Am I missing anything here?

**Reply by VictorTomaili** [reply]
service lookup editor uses the list endpoint of the related row. and can be used in different forms. When saved, the id field as foreign key saved to the current table. Works same as lookups but async and service based.

**Reply by EdwinThomas11** [reply]
Thank you for responding. I'll give this a try.

**Reply by EdwinThomas11** [reply]
@VictorTomaili I experimented with the solution provided, but despite its effectiveness, I encountered a delay of nearly 3 seconds when initializing my dialog constructor superclass. 

This delay is primarily due to the large number of properties—approximately 250—in my form file, spread across multiple tabs and categories. My objective is to significantly diminish the loading time for improved efficiency. 

You help here in greatly appreciated.

**Reply by VictorTomaili** [reply]
you can use https://github.com/serenity-is/Serenity/blob/master/packages/corelib/src/ui/helpers/lazyloadhelper.ts
initialize the tabs separately instead use them in form tab attribute

**Reply by EdwinThomas11** [reply]
@VictorTomaili , I've already segregated the majority of my properties into separate forms. As a result, I now possess numerous forms, each representing distinct tabs. Within my primary tab labeled "Details," I've approximately 250 properties. Currently, only the initialisation of the details tab in the superclass is causing a delay of approximately 3 seconds.

**Reply by VictorTomaili** [reply]
watch the network and eliminate the network request as you can. Try to separate the 250 properties to more tabs.  I can't provide more support for you.

**Reply by EdwinThomas11** [reply]
I truly appreciate your guidance and assistance here. Thank you for your support.
